java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistEnumDeclaration.java

```diff
@@ -233,24 +234,35 @@ public boolean hasInternalType(String name) {
         return this.internalTypes().stream().anyMatch(f -> f.getName().endsWith(name));
     }
 
+    @Deprecated
     public SymbolReference<? extends ResolvedValueDeclaration> solveSymbol(String name, TypeSolver typeSolver) {
         for (CtField field : ctClass.getDeclaredFields()) {
             if (field.getName().equals(name)) {
                 return SymbolReference.solved(new JavassistFieldDeclaration(field, typeSolver));
             }
         }
 
-        try {
-            for (CtClass interfaze : ctClass.getInterfaces()) {
-                SymbolReference<? extends ResolvedValueDeclaration> ref = new JavassistInterfaceDeclaration(interfaze, typeSolver).solveSymbol(name, typeSolver);
-                if (ref.isSolved()) {
-                    return ref;
-                }
+        String[] interfaceFQNs = getInterfaceFQNs();
+        for (String interfaceFQN : interfaceFQNs) {
+            SymbolReference<? extends ResolvedValueDeclaration> interfaceRef = solveSymbolForFQN(name, typeSolver, interfaceFQN);
+            if (interfaceRef.isSolved()) {
+                return interfaceRef;
             }
-        } catch (NotFoundException e) {
-            throw new RuntimeException(e);
         }
 
         return SymbolReference.unsolved(ResolvedValueDeclaration.class);
     }
+
+    private SymbolReference<? extends ResolvedValueDeclaration> solveSymbolForFQN(String symbolName, TypeSolver typeSolver, String fqn) {
```

Comment:
Maybe this method can be moved to an utility class which is package protected

Reply:
Agreed that it should be deduplicated, however I would suggest taking care of that in a later stage in this issue (but a later PR). See also the comment above re: deprecated annotation. What do you think?

For full context, this talk by Sandi Metz explains how I try to approach refactoring: https://www.youtube.com/watch?v=8bZh5LMaSmE

Reply:
Ok, we can do that